src/main/java/com/purbon/search/fair/utils/BinarySearchAlphaAdjuster.java

```diff
@@ -21,123 +29,76 @@ public BinarySearchAlphaAdjuster(int n, int k, double p, double alpha) {
     }
 
     public double adjustAlpha() {
-        return adjustIterative(alpha);
+        if (this.k <= 40 && this.k>=20) {
+            this.adjustedAlpha = adjustFlatSearch();
+            return adjustedAlpha;
+        } else {
+            this.adjustedAlpha = adjustIterative();
+            return adjustedAlpha;
+        }
+    }
+
+    private double adjustFlatSearch() {
```

Comment:
can you add a some comments, on this function? this will be very handy when we revisit this code.